src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/VisualStudioWorkspace_OutOfProc.cs

```diff
@@ -53,6 +57,25 @@ public void WaitForAsyncOperations(string featuresToWaitFor, bool waitForWorkspa
         public void WaitForAllAsyncOperations(params string[] featureNames)
             => _inProc.WaitForAllAsyncOperations(featureNames);
 
+        public void WaitForNewProjectVersion(string versionStamp, TimeSpan timeout)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            while (true)
```

Comment:
Don't we have a helper for the "retry this for a certain amount of time" helper?

Reply:
Not really. The retry helper specifically targets the known UI automation failure path and the timeout is handled outside it:

https://github.com/dotnet/roslyn/blob/88801fa234db8f1bd16b30c04971311915a04cd7/src/VisualStudio/IntegrationTest/TestUtilities/LightBulbHelper.cs#L23